### ISSUE TITLE
running recursion code will break the node process.

### ISSUE THREAD

**Issue by chejianchao** [laverdet/isolated-vm#82]
`function test(){
    test();
}

test();
`
It will break the node process and raise Bus error: 10

node version: 10.0.0 ~10.12.0,   isolated-vm version:1.7.2

**Comment by chejianchao** [comment]
I run your code, it will be ok if use Sync api to run it, but it will be broken if I use async api to run it. My OS is Mojave, 8th i7 x86-64, node 10.11.0,  llvm 10.0.0 clang-1000.11.45.2, .

**Comment by laverdet** [comment]
Can you please post the code that you're running?? I need the whole program
not just the body of the inner script.

**Comment by chejianchao** [comment]
Ok, here is the code.
```
let ivm = require('isolated-vm');
async function test(){
let isolate = new ivm.Isolate({memoryLimit:1000});
let context = await isolate.createContext();
let script  = await isolate.compileScript("function test(){test()}; test() ")
await script.run(context);
}

test()
```

**Comment by laverdet** [comment]
Ok thanks for the script. I tracked this down to the fact that OS X only allocates 512kb by default for threads, and v8 by default has a stack limit of 512kb. With the extra marginal stack overhead of pthreads and isolated-vm that puts you at risk of a real non-JS stack overflow. This is fixed in b24264c9 which will be in isolated-vm v1.7.3.

**Comment by laverdet** [comment]
The fix is now on npm, thanks for the report.
